dspace-spring-rest/src/main/java/org/dspace/app/rest/BitstreamContentRestController.java

```
@@ -9,63 +9,141 @@
+    }
+
+	@RequestMapping(method = {RequestMethod.GET, RequestMethod.HEAD})
 	public void retrieve(@PathVariable UUID uuid, HttpServletResponse response,
-            HttpServletRequest request) throws IOException {
-		BitstreamRest bit = bitstreamRestRepository.findOne(uuid);
-		if (bit == null) {
-			response.sendError(HttpServletResponse.SC_NOT_FOUND);
-			return;
+											 HttpServletRequest request) throws IOException, SQLException {
+
+		Context context = ContextUtil.obtainContext(request);
+
+        Bitstream bit = getBitstreamIfAuthorized(context, uuid, response);
+        if (bit == null) {
+            //The bitstream was not found or we're not authorized to read it.
+            return;
+        }
+
+        Long lastModified = bitstreamService.getLastModified(bit);
+        String mimetype = bit.getFormat(context).getMIMEType();
+
+		// Pipe the bits
+		try(InputStream is = getInputStream(context, bit)) {
+
+			MultipartFileSender sender = MultipartFileSender
+					.fromInputStream(is)
+					.withBufferSize(bufferSize)
+					.withFileName(bit.getName())
+					.withLength(bit.getSize())
+					.withChecksum(bit.getChecksum())
+					.withMimetype(mimetype)
+					.withLastModified(lastModified)
+					.with(request)
+					.with(response);
+
+			if (sender.isValid()) {
+				sender.serveResource();
+			}
+
+			if (sender.isNoRangeRequest() && isNotAnErrorResponse(response)) {
+				//We only log a download request when serving a request without Range header. This is because
+				//a browser always sends a regular request first to check for Range support.
+				eventService.fireEvent(
+						new UsageEvent(
+								UsageEvent.Action.VIEW,
+								request,
+								context,
+								bit));
+			}
+
+		} catch(IOException ex) {
+			log.debug("Client aborted the request before the download was completed. Client is probably switching to a Range request.", ex);
+		} catch (Exception e) {
```

No please, avoid catch all block. If we have other not runtime exceptions they should be encapsulated in a REST Exception to have single trasversal code to deal with logging and response code. We should move to the SpringMVC exception handling way https://spring.io/blog/2013/11/01/exception-handling-in-spring-mvc

---

Ok